core/src/main/java/org/neo4j/ogm/session/Neo4jSession.java

```diff
@@ -495,6 +497,48 @@ public Transaction beginTransaction(Transaction.Type type, Iterable<String> book
         return graphCallback.apply(requestHandler(), getTransaction(), metaData);
     }
 
+    /**
+     * For internal use only. Opens a new transaction if necessary before running statements
+     * in case an explicit transaction does not exist. It is designed to be the central point
+     * for handling exceptions coming from the DB and apply commit / rollback rules.
+     * @param cb The callback to execute.
+     * @param <T> The result type.
+     * @param txType Transaction type, readonly or not.
+     * @return The result of the statement.
+     */
+    public <T> T doInTransaction(Function<Transaction, T> cb, Transaction.Type txType) {
```

Comment:
1. AutoIndexManager, RequestExecutor and DeleteDelegate return always null
  additional method that wraps the return and has a void return?
2. transaction object not used in implementation of `Function`
  method signature change to e.g. (where `cb` should than be `tf` or something alike)
  `public <T> T doInTransaction(TransactionFunction<T> cb, Transaction.Type txType)`
  with 
  `public interface TransactionFunction<T>  {
      T apply();
  }`
  and removal of `transaction` parameter in `cb.apply();`